mfbt/Assertions.h

```diff
@@ -50,7 +50,21 @@ MOZ_BEGIN_EXTERN_C
 extern MFBT_DATA const char* gMozCrashReason;
 MOZ_END_EXTERN_C
 
-#if defined(MOZ_HAS_MOZGLUE) || defined(MOZILLA_INTERNAL_API)
+#if defined(__HAIKU__)
+/*
+ * We introduce a function prototype here as including Haiku specific header
+ * from this globally included header will cause too many conflicts.
+ */
+MOZ_BEGIN_EXTERN_C
+// Workaround for gcc_hidden.h hack of libxul.so
+#pragma GCC visibility push(default)
+extern void debugger(const char *message);
+#pragma GCC visibility pop
+MOZ_END_EXTERN_C
+
+#  define MOZ_CRASH_ANNOTATE(...) debugger(__VA_ARGS__)
+
+#elif defined(MOZ_HAS_MOZGLUE) || defined(MOZILLA_INTERNAL_API)
 static inline void AnnotateMozCrashReason(const char* reason) {
   gMozCrashReason = reason;
```

Comment:
Original code will be used only MOZ_CRASHREPORTER defined. Non tier-1 platforms doesn't support this.

https://searchfox.org/mozilla-esr128/source/toolkit/moz.configure#3156

So altering this looks safe.